multi_imbalance/resampling/MDO.py

```diff
@@ -0,0 +1,110 @@
+from random import sample
+from collections import Counter
+
+from sklearn.decomposition import PCA
+from sklearn.neighbors import NearestNeighbors
+
+import numpy as np
+
+
+class MDO(object):
+    """
+    Mahalanbois Distance Oversampling is an algorithm that oversamples all classes to a quantity of the major class.
+    Samples for oversampling are chosen based on their k neighbours and new samples are created in random place but
+    with the same Mahalanbois distance from the centre of class to chosen sample.
+
+    """
+
+    def __init__(self, k=9, k1_frac=.5):
+        self.knn = NearestNeighbors(n_neighbors=k)
+        self.k2 = k
+        self.k1 = int(k * k1_frac)
+
+    def fit_transform(self, X, y):
+        """
+
+        Parameters
+        ----------
+        X two dimensional numpy array (number of samples x number of features) with float numbers
+        y one dimensional numpy array with labels for rows in X
+
+        Returns
+        -------
+        Resampled X and y
+        """
+        self.knn.fit(X)
+        oversampled_X, oversampled_y = X.copy(), y.copy()
+        quantities = Counter(y)
+        goal_quantity = int(max(list(quantities.values())))
+
+        labels = list(set(y))
+        for class_label in labels:
+            SC_minor, weights = self._choose_samples(X, y, class_label)
+            if (len(SC_minor)) == 0:
```

Comment:
I don't think that these extra brackets are necessary, do you?

Reply:
done